### PR TITLE
Clean up constants

### PR DESCRIPTION
**Pull request by od0x0**
I mostly did this as a pipecleaner for the new workflow, but it scratches an itch I think we've all expressed once or twice.

There's now a single constant node with a payload of enum type "Constant".
This should stop the proliferation of constant nodes for every type that needs a slightly different payload encoding to be type safe (and not just rely on casts of the binary data).

### CODE REVIEW

**Comment by Checkmate50** [comment]
Looks good to me, exactly the changes I would expect.  Looks like I'll need to fix up the assembly a bit, just as a note.

**Review by Checkmate50 — APPROVED** [review]

**Review by mjguynn — APPROVED** [review]
Looks good to me!

**Review by miahdan — APPROVED** [review]
Looks good!

File: caiman-spec/src/spec.rs [hunk 1]
```diff
@@ -33,6 +33,7 @@ pub enum OperationInputKind
 {
 	Type,
 	Place,
+	Immediate,
 	ImmediateI64,
 	ImmediateI32,
 	ImmediateU64,
```
[the remark this answers is not in the document]
Comment (od0x0) [reply caiman-spec/src/spec.rs:39]: I think I'll wait a bit for this. We might soon have some instructions that will only want an immediate of a specific type and having to go through a constant might complicate them.

**Pull request merged**